File: 백준/Silver/2559. 수열/수열.cc
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int n, k;

int main() {
    cin.tie(NULL);
    ios::sync_with_stdio(false);
    
    cin >> n >> k;
    
    vector<int> arr(n, 0);
    vector<int> sum(n+1, 0);
    
    for(int i = 0; i < n; i++){
        cin >> arr[i];
        sum[i+1] = arr[i];
    }
    for(int i = 1; i <= n; i++){
        sum[i] += sum[i-1];
    }
    
    int res = 100000*-100;
    for(int i = k; i <= n; i++){
        res = max(res, sum[i] - sum[i-k]);
    }
    cout << res;
}

File: 백준/Silver/2559. 수열/수열.java
import java.util.*;
import java.lang.*;
import java.io.*;

class Main {
	public static void main (String[] args) throws java.lang.Exception {
	   BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
	   String[] tmps = br.readLine().split(" ");
	   String[] arr = br.readLine().split(" " );
	   int[] sum = new int[arr.length+1];
	   int n = Integer.parseInt(tmps[0]);
	   int k = Integer.parseInt(tmps[1]);
	   
	   for(int i = 1; i <= n; i++){
	       sum[i] = Integer.parseInt(arr[i-1]);
	       sum[i] += sum[i-1];
	   }
	   
	   int res = Integer.MIN_VALUE;
	   
	   for(int i = k; i <= n; i++){
	       res = Math.max(res, sum[i] - sum[i-k]);
	   }
	   System.out.print(res);
	}
}